lib/cfa/matcher.rb

```diff
@@ -1,9 +1,32 @@
 module CFA
-  # Class used to create matcher, that allows to find specific option in augeas
-  # tree or subtree
-  # TODO: examples of usage
+  # The Matcher is used as a predicate on {AugeasElement}.
```

Comment:
in fact I would like to have generic matcher here and if more parsers added, then I like to make it universalally used. So now it is tight with AugeasElement only because it is only parser.